Added optimiser macros and modified dsp_optimize to work with init args

SeanSullivan1 opened the pull request.

Commit:
Added optimiser macros and modified dsp_optimize to work with init args

ggmarshall commented on pygama/genpar_tmp/optimiser_macro.py:
```diff
@@ -0,0 +1,52 @@
+    string_values = [ f'{val:.2f}{par_values["unit"]}' for val in string_values]
+    return string_values
+
+def build_energy_cuts(file,Emin,Emax):
```
I think it would be better to change this function to output a mask as this will be much easier to combine with other cuts (also simplifies the code) something like mask = (wf_maxes>Emin)&(wf_maxes<Emax)

SeanSullivan1 replied:
Changed

Commit:
Added docstrings for main optimiser macro functions
Changed build_energy_cuts to build_energy_mask, returning mask array
Changed fom_FWHM to use existing pygama histogram functions
